### PR TITLE
Add auth factor methods

### PR DESCRIPTION
**Pull request by blairlunceford**
## Description

Adds methods to enroll an authentication factor and list the authentication factors for a user.

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: workos/resources/mfa.py [hunk 1]
```diff
@@ -115,3 +115,51 @@ def to_dict(self):
         verification_response_dict = super(WorkOSChallengeVerification, self).to_dict()
 
         return verification_response_dict
+
+
+class WorkOSAuthenticationChallengeAndFactor(WorkOSBaseResource):
```
Comment (amygdalama) [inline workos/resources/mfa.py:120]: i would recommend naming this something like `EnrollAuthFactorResponse` instead (existing example: https://github.com/workos/workos-python/blob/fb0e087584c47f3fdc3ac29e137733b59fe2a15a/workos/resources/password_challenge_response.py#L5 )

or, leaving as two separate classes and building the response in `enroll_auth_factor` (this is what the node SDK does: https://github.com/workos/workos-node/blob/c46747ba60114bed4e0c091a3df7e97ce5a6c894/src/user-management/user-management.ts#L274)

File: workos/resources/mfa.py [hunk 2]
```diff
@@ -115,3 +115,51 @@ def to_dict(self):
...
+
+        authentication_challenge_and_factor.authentication_factor = (
+            WorkOSAuthenticationFactorTotp.construct_from_response(
+                response["authentication_factor"]
+            )
+        )
+
+        return authentication_challenge_and_factor
+
+    def to_dict(self):
```
Comment (amygdalama) [inline workos/resources/mfa.py:152]: do we use this anywhere? seems like we'd only need to go from json -> response object for this case, and not object -> json?

File: workos/user_management.py [hunk 3]
```diff
@@ -540,3 +541,72 @@ def send_magic_auth_code(
...
+            token=workos.api_key,
+        )
+
+        return WorkOSAuthenticationChallengeAndFactor.construct_from_response(
+            response
+        ).to_dict()
+
+    def list_auth_factors(
+        self,
+        user,
```
Comment (amygdalama) [inline workos/user_management.py:589]: do we tend to use params like `user` or `user_id` in the python SDK more? what about within the user management module?

File: workos/resources/mfa.py (hunk 2 above)
Comment (amygdalama) [reply workos/resources/mfa.py:152]: ohh i see we use `to_dict` down here, i missed it https://github.com/workos/workos-python/pull/217/files#diff-87ef5545759351bbeeba22da6f30add44d7c735f751c4807fd059d6ef87931c6R585

wat? lol

Comment (amygdalama) [reply workos/resources/mfa.py:152]: ok, i think i see what is happening. we are going from dict -> Response object -> dict. i assume to control the shape of the dict we return from `enroll_auth_factor` ?

Comment (amygdalama) [reply workos/resources/mfa.py:152]: it's certainly weird but matches what we currently do in the python SDK so ¯\\_(ツ)_/¯

Comment (blairlunceford) [reply workos/resources/mfa.py:152]: Yeah I'm not sure where the pattern derives from, but I just followed what I saw in other resources.

File: workos/user_management.py (hunk 3 above)
Comment (blairlunceford) [reply workos/user_management.py:589]: It seems to be more like `user`, `connection`, `organization`, etc. It definitely seems like doing the `_id` would be better, but that is the convention we're currently using elsewhere:

https://github.com/workos/workos-python/blob/fb0e087584c47f3fdc3ac29e137733b59fe2a15a/workos/user_management.py#L75

https://github.com/workos/workos-python/blob/fb0e087584c47f3fdc3ac29e137733b59fe2a15a/workos/sso.py#L175

https://github.com/workos/workos-python/blob/fb0e087584c47f3fdc3ac29e137733b59fe2a15a/workos/organizations.py#L157

Comment (amygdalama) [reply workos/user_management.py:589]: and we use organization_id here... but that's a list endpoint https://github.com/workos/workos-python/blob/fb0e087584c47f3fdc3ac29e137733b59fe2a15a/workos/user_management.py#L93

Comment (amygdalama) [reply workos/user_management.py:589]: blegh! lol

File: workos/resources/mfa.py (hunk 1 above)
Comment (blairlunceford) [reply workos/resources/mfa.py]: I went with the Node example 👍

File: workos/user_management.py (hunk 3 above)
Comment (blairlunceford) [reply workos/user_management.py:597]: yikes 🙈 I'm ok with starting to be more opinionated and go the `*_id` route

File: workos/resources/mfa.py (hunk 1 above)
Comment (amygdalama) [reply workos/resources/mfa.py]: beautiful 🎨

File: workos/user_management.py (hunk 3 above)
Comment (amygdalama) [reply workos/user_management.py:597]: yeah how about within the UM module we go w/ `*_id`?

fwiw we were also inconsistent at the API layer but for UM have made all params `*_id` (and that'll be the plan for new APIs going forward)

Comment (blairlunceford) [reply workos/user_management.py:597]: cool cool - I'll make those changes throughout the module now 👍

Comment (blairlunceford) [reply workos/user_management.py]: I made those changes throughout the UM module, just to remind myself for later changes

**Review by amygdalama — APPROVED** [review]
ty!

**Pull request merged**